src/main/java/de/codecentric/centerdevice/util/StageUtils.java

```diff
@@ -46,12 +56,49 @@ public static void focusNextStage() {
 		}
 	}
 
+	@SuppressWarnings("unchecked")
 	public static ObservableList<Stage> getStages() {
-		return StageHelper.getStages();
+		if (stages == null) {
+			// Java 9
+			try {
+				windows = (ObservableList<Window>)Window.class.getMethod("getWindows").invoke(null);
+				stages = FXCollections.observableArrayList();
+				windows.addListener(new ListChangeListener<Window>() {
+					@Override
+					public void onChanged(javafx.collections.ListChangeListener.Change<? extends Window> c) {
+						updateStages();
```

Comment:
I am not sure whether this is really necessary and what advantage this would have. I just tried to keep the code simple. It may make a difference if you have a lot of windows and you open and close them very frequently. But I don't have any use-case for this and thus cannot test it. I'd leave the code as it is but watch this issue for a future improvement.